jekyll/_cci2/github-apps-integration.adoc

```diff
@@ -0,0 +1,255 @@
+
+In addition to a configuration source, each trigger includes the webhook URL, and in this scenario, a CircleCI-created GitLab token. The webhook URL and GitLab token are used to securely register the webhook within GitLab in order to receive push events from your GitLab repo.
+
+image::{{site.baseurl}}/assets/img/docs/gl-ga/gitlab-ga-project-settings-edit-trigger.png[Trigger details]
+
+**Trigger filters** allow you to determine when a trigger should initiate a build based on the parameters provided by Gitlab’s webhook. CircleCI provides some common options, for example, only build on merge requests, but you can also build your own rules using the custom filter option. For example, a custom filter would allow you to only build on a specific branch or user.
+
+image::{{site.baseurl}}/assets/img/docs/gl-preview/gitlab-preview-project-settings-customize-triggers.png[Trigger details]
+
+NOTE: Currently, the only fields that can be edited for existing filters are **Trigger Name** and the **Filters** radio buttons.
+
+[#project-settings-advanced]
+=== Advanced
+
+- You can enable dynamic configuration using setup workflows in CircleCI. To learn about dynamic configuration, read the xref:dynamic-config#[Dynamic configuration] guide.
+- At this time, the **Free and Open Source** setting is not currently supported, but there are plans to make this available in the future.
+- At this time, auto-cancel redundant workflows is not supported. Refer to the xref:skip-build#auto-cancelling[Auto cancelling] section of the `skip` or `cancel` jobs and workflows page for more details.
+
+[#project-settings-ssh-keys]
+=== GitHub project SSH keys
+
+When creating a project in CircleCI, you will create and add SSH keys, which are used to check out code from your repo. Each configuration you create generates a new SSH key to access the code in the repo associated with that configuration. At this time, only **Additional SSH Keys** are applicable to GitHub App integrations.
+
+[#create-ssh-key]
+==== Create an SSH key
+
+You will be guided through the SSH key generation processing in the CircleCI web app. The steps are as follows:
+
+. Create an SSH key-pair by using the following command. When prompted to enter a passphrase, do **not** enter one:
++
+```shell
+  ssh-keygen -t ed25519 -C "your_email@example.com"
+```
+
+. Go to your project on link:https://gitlab.com/[GitLab] and navigate to **Settings > Repository**, and expand the **Deploy keys** section. Enter a title in the "Title" field, then copy and paste the public key you created in step 1. Check **Grant write permissions to this key** box, then click **Add key**.
+
+. Go to your project settings in the CircleCI app, select **SSH Keys**, and **Add SSH key**. In the "Hostname" field, enter `gitlab.com` and add the private key you created in step 1. Then click **Add SSH Key**.
```

Comment:
We'll want to note that the input should include the ---BEGIN--- and ---END--- part of the SSH key